src/test/assembly/asm/hexagon-types.rs

```diff
@@ -0,0 +1,144 @@
+// no-system-llvm
+// assembly-output: emit-asm
+// compile-flags: --target hexagon-unknown-linux-musl
+
+#![feature(no_core, lang_items, rustc_attrs, repr_simd)]
+#![crate_type = "rlib"]
+#![no_core]
+#![allow(asm_sub_register, non_camel_case_types)]
+
+#[rustc_builtin_macro]
+macro_rules! asm {
+    () => {};
+}
+#[rustc_builtin_macro]
+macro_rules! stringify {
+    () => {};
+}
+
+#[lang = "sized"]
+trait Sized {}
+#[lang = "copy"]
+trait Copy {}
+
+type ptr_8 = *const i8;
+type ptr_16 = *const i16;
+type ptr_32 = *const i32;
+type ptr_mut_32 = *mut i32;
```

Comment:
You only need to test one pointer type here, they are all treated identically by the backend.